kalagraphs.module

```diff
@@ -223,24 +238,39 @@ function _kalagraphs_populate_fields(&$vars) {
     return;
   }
 
+  $parent = $paragraph->getParentEntity();
+  $author = $parent->getOwner();
+
   // Add a flag to indicate it is rendering in a Drupal context.
   // @todo This doesn't cover non-Paragraph theme hooks, e.g., atom components
   // used for some fields.
   $vars['drupal'] = TRUE;
 
+  // Set generic values
+  $vars['author'] = $author->getUsername();
+  $date = DrupalDateTime::createFromTimestamp($parent->getCreatedTime());
+  $vars['date'] = $date->format('m/d/Y - h:i A');
+
   // Map field items to template variables.
   foreach (Element::children($elements) as $field) {
+    // Make a template-friendly variable name by stripping "field_kalagraphs_"
+    // from the front.
+    $name = substr($field, 17);
     $element = &$elements[$field];
+    $value = isset($element['#items']) ? $element['#items']->getValue() : array();
     foreach (Element::children($element) as $item) {
-      // Make a template-friendly variable name by stripping "field_kalagraphs_"
-      // from the front.
-      $name = substr($field, 17);
-      $vars[$name][] = $element[$item];
+      $vars[$name][$item] = $element[$item];
+      if ($name == 'text' && isset($value[$item]['summary'])) {
+        $vars['subtitle'] = $value[$item]['summary'];
+      }
     }
   }
 
   // Provide a unique id attribute for the element that may or may not get used.
   $vars['id'] = Html::getUniqueId("kalagraphs-component-{$paragraph->field_kalagraphs_type->value}");
+  
+  // Force callback of the preprocess hook
+  \Drupal::moduleHandler()->invokeAll("preprocess_kalagraphs_{$paragraph->field_kalagraphs_type->value}", [&$vars]);
```

Comment:
Seems that way, but we have different ideas in mind :) Hooks are dead! Take this bit out and then it's good to merge, the rest looks really good.